tests/test_mpi.py

```diff
@@ -180,6 +180,44 @@ def test_custom_topology(self):
         assert f2.shape == expected[distributor.myrank]
         assert f2.size_global == f.size_global
 
+    @pytest.mark.parametrize('topology, dist_topology, expected', [
+        (('*', 1, 1), (4, 1, 1), [(4, 15, 15), (4, 15, 15), (4, 15, 15), (3, 15, 15)]),
+        ((1, '*', 1), (1, 4, 1), [(15, 4, 15), (15, 4, 15), (15, 4, 15), (15, 3, 15)]),
+        ((1, 1, '*'), (1, 1, 4), [(15, 15, 4), (15, 15, 4), (15, 15, 4), (15, 15, 3)]),
+        (('*', 1, '*'), (2, 1, 2), [(8, 15, 8), (8, 15, 7), (7, 15, 8), (7, 15, 7)]),
+    ])
+    @pytest.mark.parallel(mode=[4])
```

Comment:
Can we test the class CustomTopology "in isolation", that is without launching an actual MPI test? MPI tests are slow (because they fork new processes etc), so if avoidable it would be better. Here they should be avoidable with some expedients

Reply:
As far as I am aware, all the tests that use a Distributor are marked parallel, i.e. launching jobs.
Do we have an example of not doing that?

Reply:
I can sketch something but not sure I like it or want it.
Also I believe it is worth to have it running w mpi, but up to you.

Reply:
I pushed an idea with a dummy communicator class, but only for the custom topology.
The current Distributor implementation resorts to (1,1,1) in case of a non-MPI init communicator present.

Reply:
> As far as I am aware, all the tests that use a Distributor are marked parallel, i.e. launching jobs.
Do we have an example of not doing that?

yes in this file IIRC

the new test with the dummy communicator is exactly what I had in mind. It's lightweight and it achieves the objective of the test we need to add